src/main/java/org/jboss/ejb/client/EJBRootContext.java

```diff
@@ -83,9 +83,15 @@ protected Object lookupNative(final Name name) throws NamingException {
             }
         }
         if (beanName == null) {
+            if (size == 3) {
+                // name is of the form appName/moduleName/distinctName
+                return new RelativeContext(new FastHashtable<>(getEnvironment()), this, SimpleName.of(name));
+            }
```

Comment:
I'm trying to work out now if there's some case where the bean name is null where it wasn't meant to be; I guess that must be what's happening?